### PR TITLE
Fix Assistant API Vision Capabilities

### PR DESCRIPTION
**Pull request by keepingitneil**

### CODE REVIEW

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py [hunk 1]
```diff
@@ -162,13 +176,15 @@ def chat(
         if not self._sync_openai_task:
             self._sync_openai_task = asyncio.create_task(self._sync_openai())
 
+        print("NEIL chatting with", chat_ctx)
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py:179]: nit: log

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py [hunk 2]
```diff
@@ -360,14 +409,18 @@ async def _create_stream(self) -> None:
                 "event_handler": eh,
                 "temperature": self._temperature,
             }
+            print("NEIL kwargs", kwargs)
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py:412]: same

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py [hunk 3]
```diff
@@ -360,14 +409,18 @@ async def _create_stream(self) -> None:
...
                     for f in self._fnc_ctx.ai_functions.values()
                 ]
 
-            async with self._client.beta.threads.runs.stream(**kwargs) as stream:
-                await stream.until_done()
+            try:
+                async with self._client.beta.threads.runs.stream(**kwargs) as stream:
+                    await stream.until_done()
+            except Exception as e:
+                print("NEIL error", e)
```
Comment (davidzhao) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py:423]: how should we handle errors here?

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py [hunk 4]
```diff
@@ -81,6 +93,7 @@ def __init__(
         client: AsyncClient | None = None,
         api_key: str | None = None,
         base_url: str | None = None,
+        on_file_uploaded: OnFileUploaded | None = None,
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py:96]: Let's follow the callback we have in VoiceAssistant, what about file_uploaded_cb ?

File: livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py [hunk 5]
```diff
@@ -413,3 +490,53 @@ async def __anext__(self):
             raise item
 
         return item
+
+
+def build_oai_message(msg: llm.ChatMessage, cache_key: Any):
```
Comment (theomonnom) [inline livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/beta/assistant_llm.py:495]: Do we have to copy this function?

**Review by theomonnom — APPROVED** [review]
Otherwise lgtm.

**Pull request merged**